### PR TITLE
Improving tests

### PR DESCRIPTION
**Pull request by giovannicuriel**
This PR adds new tests and improves a few old ones. 
Also, it reduces the total of tests executed by Gruntfile so that only those somewhat useful for mashup are actually executed.

This PR is related to dojot/dojot#58

### CODE REVIEW

File: orchestrator/translator.js [hunk 1]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
+        expression.coords = '';
+        for (let i = 0; i < ruleValue.length; i++) {
+          let point = ruleValue[i];
+          expression.coords += point.latitude + ',' + point.longitude + ';';
+        }
+        // Closing the polygon
+        expression.coords += ruleValue[0].latitude + ',' + ruleValue[0].longitude;
+        eventConditionArray.push(expression);
+      } else {
+        ret.status = 'invalid geofence mode';
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js [hunk 2]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
+        ret.code = -1;
       }
-      // Closing the polygon
-      expression.coords += ruleValue[0].latitude + ',' + ruleValue[0].longitude;
-      eventConditionArray.push(expression);
     }
+  } else {
+    ret.status = 'invalid operator';
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js [hunk 3]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
-        'coords': '',
-        'geometry': ''
-      };
-      expression.geometry = orchtypes.OrionTypes.GeoFenceMode.POLYGON;
-      expression.coords = '';
-      for (let i = 0; i < ruleValue.length; i++) {
-        let point = ruleValue[i];
-        expression.coords += point.latitude + ',' + point.longitude + ';';
+    if (ruleValue.length == 0) {
+      ret.status = 'empty georeference node';
```
Comment (mmagr) [inline orchestrator/translator.js]: can we add the id of the invalid node in the error object? (that way we might help the user identify the invalid node)

File: orchestrator/translator.js [hunk 4]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
 
 function addEventCondition(node, ruleOperation, ruleValue, ruleType, request, eventConditionArray) {
+  let ret = {
+    'status' : 'ok',
+    'code': 0
+  }
+  // Sanity checks
+  for (let i = 0; i < arguments.length; i++) {
+    if (arguments[i] == undefined || arguments[i] === null) {
+      ret.status = 'invalid parameter';
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js [hunk 5]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
-          // This is this node!
-          break;
         default:
           addFixedEventCondition(node, orchtypes.NodeRed.NegatedLogicalOperators[ruleOperation], ruleValue, ruleType, request);
       }
+    } else {
+      if (ruleOperation in orchtypes.NodeRed.LogicalOperators) {
+        ret.status = 'operator has no negated form';
+      } else {
+        ret.code = -1;
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js [hunk 6]
```diff
@@ -36,6 +36,20 @@ function isValidGeoOperator(op) {
...
  */
 function addNegatedFixedEventCondition(node, request) {
+  let ret = {
+    'status' : 'ok',
+    'code': 0
+  }
+  // Sanity checks
+  for (let i = 0; i < arguments.length; i++) {
+    if (arguments[i] == undefined || arguments[i] === null) {
+      ret.status = 'invalid parameter';
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js [hunk 7]
```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
...
+
+  let ret = {
+    'code': 0,
+    'status': 'ok',
+    'requestList': []
+  }
+
+  for (let i = 0; i < arguments.length; i++) {
+    if ((arguments[i] == null) || (arguments[i] == undefined)) {
+      ret.code = -1;
```
Comment (mmagr) [inline orchestrator/translator.js]: ... and here

File: orchestrator/translator.js (hunk 3 above)
Comment (giovannicuriel) [reply orchestrator/translator.js]: Sure!

**Review by mmagr — APPROVED** [review]
Looks good.

**Pull request merged**